raysect/core/math/function/function1d/base.pxd

```diff
@@ -95,3 +96,13 @@ cdef class PowScalarFunction1D(Function1D):
 cdef class PowFunctionScalar1D(Function1D):
     cdef double _value
     cdef Function1D _function
+
+
+cdef inline bint is_callable(object f):
+    """
+    Tests if an object is a python callable or function object.
+    
+    :param object f: Object to test.  
+    :return: True if callable, False otherwise.
+    """
+    return isinstance(f, Function1D) or callable(f)
```

Comment:
Is there are set rule for newlines at the end of files? Other files in the project seem to have them, and my editor adds them automatically.